plugins/callback/loganalytics.py

```diff
@@ -0,0 +1,267 @@
+
+    def _runtime(self, result):
+        return (
+            datetime.utcnow() -
+            self.start_datetimes[result._task._uuid]
+        ).total_seconds()
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.workspace_id = self.get_option('workspace_id')
+
+        if self.workspace_id is None:
+            self.disabled = True
+            self._display.warning('Azure Log analytics workspace ID was '
+                                  'not provided. The workspace '
+                                  'Workspace ID can be provided using the '
+                                  '`WORKSPACE_ID` environment variable or '
+                                  'in the ansible.cfg file.')
+
+        self.shared_key = self.get_option('shared_key')
+
+        if self.shared_key is None:
+            self.disabled = True
+            self._display.warning('Azure Log analytics workspace requires a '
+                                  'shared key. The authentication shared key '
+                                  'can be provided using the '
+                                  '`WORKSPACE_SHARED_KEY` environment variable or '
+                                  'in the ansible.cfg file.')
+
+    def v2_playbook_on_play_start(self, play):
+        vm = play.get_variable_manager()
+        extra_vars = vm.extra_vars
+        self.loganalytics.extra_vars = extra_vars
+
+    def v2_playbook_on_start(self, playbook):
+        self.loganalytics.ansible_playbook = basename(playbook._file_name)
+
+    def v2_playbook_on_task_start(self, task, is_conditional):
+        self.start_datetimes[task._uuid] = datetime.utcnow()
+
+    def v2_playbook_on_handler_task_start(self, task):
+        self.start_datetimes[task._uuid] = datetime.utcnow()
+
+    def v2_runner_on_ok(self, result, **kwargs):
+        self.loganalytics.send_event(
+            self.workspace_id,
+            self.shared_key,
+            'OK',
+            result,
+            self._runtime(result)
+        )
+
+    def v2_runner_on_skipped(self, result, **kwargs):
+        self.loganalytics.send_event(
+            self.workspace_id,
+            self.shared_key,
+            'SKIPPED',
+            result,
+            self._runtime(result)
+        )
+
+    def v2_runner_on_failed(self, result, **kwargs):
+        self.loganalytics.send_event(
+            self.workspace_id,
+            self.shared_key,
+            'FAILED',
+            result,
+            self._runtime(result)
+        )
+
+    def runner_on_async_failed(self, result, **kwargs):
+        self.loganalytics.send_event(
+            self.workspace_id,
+            self.shared_key,
+            'FAILED',
+            result,
+            self._runtime(result)
+        )
+
+    def v2_runner_on_unreachable(self, result, **kwargs):
```

Comment:
Add a changelog entry [here](https://github.com/ansible-collections/community.general/tree/main/changelogs/fragments) and test for this change.

Reply:
@Akasurde thank you Abhijeet. Changelog fragment is added, I will add some tests tomorrow after work.

Reply:
@felixfontein @Akasurde added unit test test_loganalytics.py